src/Illuminate/Http/TransformableResponse.php

```diff
@@ -0,0 +1,335 @@
+<?php
+
+namespace Illuminate\Http;
+
+use Illuminate\Support\Arr;
+use Illuminate\Support\Str;
+use Illuminate\Validation\ValidationData;
+use Illuminate\Contracts\Support\Arrayable;
+
+abstract class TransformableResponse extends JsonResponse
+{
+    /**
+     * Constructor.
+     *
+     * @param  mixed  $data
+     * @param  int  $status
+     * @param  array  $headers
+     * @param  int  $options
+     * @return void
+     */
+    public function __construct($data = null, $status = 200, $headers = [], $options = 0)
+    {
+        if (is_array($data)) {
```

Comment:
> How we should treat \Illuminate\Contracts\Support\Jsonable ??
Have a look at `\Illuminate\Routing\Router::prepareResponse`

> How do we should treat a Collection of Transformables
If data is a collection, map all its items

> Should Transformable prevail against Arrayable
I believe so